src/containers/GalleryWidget/GalleryWidget.jsx

```diff
@@ -1,49 +1,72 @@
 import React, { Component } from 'react';
 import axios from 'axios';
 import './GalleryWidget.scss';
+import LoadingSpinner from '../../components/LoadingSpinner';
+import { tenMinutes, tenSeconds } from '../../helper/DateUtils';
 
 class GalleryWidget extends Component {
-
   constructor(props) {
     super(props);
 
     this.state = {
-      urls: [],
-      currentImage: 0
+      loading: true,
+      images: [],
+      currentImage: {}
     };
 
     this.getData = this.getData.bind(this);
-    this.getRandomImageIndex = this.getRandomImageIndex.bind(this);
-
-    this.getData();
+    this.rotateCurrentImage = this.rotateCurrentImage.bind(this);
   }
 
   componentDidMount() {
-    this.interval = setInterval(this.getRandomImageIndex, 30000);
-    this.interval = setInterval(this.getData, 1000 * 60 * 10);
+    this.getData();
+
+    this.interval = setInterval(this.getData, tenMinutes());
+    this.timer = setInterval(this.rotateCurrentImage, tenSeconds());
   }
 
   getData() {
-    axios.get(this.props.api).then((urls) => this.setState({ urls : urls.data }));
+    axios.get(this.props.api).then(images => {
+      this.setState({ loading: false, images: images.data });
+      // Force Rotate
+      this.rotateCurrentImage();
+    });
   }
 
-  getRandomImageIndex() {
-    let index = Math.floor(Math.random() * this.state.urls.length);
-    this.setState({ currentImage: index });
+  async rotateCurrentImage() {
```

Comment:
The problem with it is that it was picking the same image since there's a chance that it'll display the same image and I didn't quite like it so i think iterating in order makes it cleaner.

Reply:
Fair enough, I'll just jumble around the images on the drive.